rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala

```diff
@@ -0,0 +1,410 @@
+/*
+*************************************************************************************
+* Copyright 2018 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.db
+
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets
+import java.sql.Timestamp
+import java.util.Properties
+
+import com.normation.rudder.db.Doobie.{DateTimeMeta, slf4jDoobieLogger}
+import com.normation.ldap.sdk.BuildFilter
+import com.normation.ldap.sdk.ROPooledSimpleAuthConnectionProvider
+import com.normation.ldap.sdk.RWPooledSimpleAuthConnectionProvider
+import com.normation.rudder.db.Doobie.query
+import com.normation.rudder.domain.NodeDit
+import com.normation.rudder.domain.RudderDit
+import com.normation.rudder.domain.policies.RuleId
+import com.normation.rudder.domain.reports.ComplianceLevel
+import com.normation.rudder.repository.jdbc.RudderDatasourceProvider
+import doobie.imports._
+import scalaz._
+import scalaz.Scalaz._
+import doobie.postgres._
+import doobie.postgres.imports._
+import com.unboundid.ldap.sdk.DN
+import net.liftweb.common.Box
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Full
+import org.joda.time.DateTime
+import org.joda.time.Interval
+
+import scala.util.Random
+
+
+/*
+ * Allow to generate false compliance for testing purpose
```

Comment:
At last, A service for "All green" reporting mode ;)

Reply:
not even! it randomize output, so when you generate sufficiently big number of reports, you get an almost flat compliance

Reply:
Yes i have seen but it's the first step to that audit mode needed by all Rudder users !!